samples/workload/tensorflow/tflite_bazel_building.patch

```diff
@@ -0,0 +1,791 @@
+--- a/tensorflow/lite/BUILD
++++ b/tensorflow/lite/BUILD
+@@ -58,6 +58,15 @@ config_setting(
+     },
+ )
+ 
++config_setting(
++    name = "emscripten_wasmsimd",
++    values = {
++        "crosstool_top": "@emsdk//emscripten_toolchain:everything",
++        "cpu": "wasm",
++        "features": "wasm_simd",
++    },
++)
++
+ # Without "cpu":"k8", when building with --copt=-DTF_LITE_STATIC_MEMORY, we get
+ # the following error:
+ # Multiple matches are not allowed unless one is unambiguously more specialized.
+diff --git a/tensorflow/lite/build_def.bzl b/tensorflow/lite/build_def.bzl
+index 842879a0536..4d1ae920c85 100644
+--- a/tensorflow/lite/build_def.bzl
++++ b/tensorflow/lite/build_def.bzl
+@@ -36,6 +36,10 @@ def tflite_copts():
+             "/wd4018",  # -Wno-sign-compare
+             # copybara:comment_end
+         ],
++        clean_dep("//tensorflow:emscripten_wasmsimd"): [
++            "-sSHARED_MEMORY=0",
++            "-sUSE_PTHREADS=0",
++        ],
+         "//conditions:default": [
+             "-Wno-sign-compare",
+         ],
+diff --git a/tensorflow/lite/mmap_allocation.cc b/tensorflow/lite/mmap_allocation.cc
+index 3d1a7f03e71..3d18778eb3a 100644
+--- a/tensorflow/lite/mmap_allocation.cc
++++ b/tensorflow/lite/mmap_allocation.cc
+@@ -12,7 +12,6 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ==============================================================================*/
+-
+ #include <fcntl.h>
+ #include <stddef.h>
+ #include <sys/mman.h>
+diff --git a/tensorflow/lite/tools/benchmark/BUILD b/tensorflow/lite/tools/benchmark/BUILD
+index 95ca87a9e87..3550f1a5ef2 100644
+--- a/tensorflow/lite/tools/benchmark/BUILD
++++ b/tensorflow/lite/tools/benchmark/BUILD
+@@ -1,6 +1,7 @@
+ load("//tensorflow:tensorflow.bzl", "tf_cc_binary")
+ load("//tensorflow/lite:special_rules.bzl", "tflite_portable_test_suite")
+ load("//tensorflow/lite:build_def.bzl", "tflite_copts", "tflite_copts_warnings", "tflite_linkopts")
++load("@emsdk//emscripten_toolchain:wasm_rules.bzl", "wasm_cc_binary")
+ 
+ package(
+     # copybara:uncomment default_applicable_licenses = ["//tensorflow:license"],
+@@ -37,6 +38,20 @@ cc_binary(
+             "-lm",  # some builtin ops, e.g., tanh, need -lm
+             "-Wl,--rpath=/data/local/tmp/",  # Hexagon delegate libraries should be in /data/local/tmp
+         ],
++        "//tensorflow:emscripten_wasmsimd": [
++            "-o benchmark_model.wasm",
++            "-s ASSERTIONS=1",
++            "-s ERROR_ON_UNDEFINED_SYMBOLS=0",
++            "-s ALLOW_MEMORY_GROWTH=1",
++            "-s TOTAL_MEMORY=536870912",  # 512M
++            "-s SHARED_MEMORY=0",
++            "-s USE_PTHREADS=0",
++            "-O0",
++            "-Wl,--export=__heap_base",
++            "-Wl,--export=__data_end",
++            "-Wl,--export=malloc",
++            "-Wl,--export=free",
++        ],
```

Comment:
After further research, I could remove this pre-allocation of 512MB of memory by removing `-s TOTAL_MEMORY=536870912`, which now works with an appropriate enclave size.

Is there any reason to set this flag?
Cheers

Reply:
`TOTAL_MEMORY` is the [initial amount of memory](https://github.com/emscripten-core/emscripten/blob/main/src/settings.js#L157C1-L165C31). The application will firstly allocate that much memory. It will grow the memory when there isn't enough memory and `ALLOW_MEMORY_GROWTH` is enabling. But the *grow* action is heavy and slow.

512MB is just a suggestion not a mandatory. you're free to change it.

Reply:
What do you think of either omitting this flag in the patch, and documenting it in the README for optimizing the performance. Alternatively, may we reduce its value so `iwasm` from `linux-sgx` may work with the workload without editing the patch?

Cheers

Reply:
we didn't tend to fix the value of `TOTAL_MEMORY`. you are totally free to replace the number. Since the value should be different with variant models or variant environments. 

I guess we'd like to highlight that point in README.md

Reply:
Okay, as long as you document this value for making it work with `linux-sgx`, you can consider this task as working with Intel SGX.